lib/Db/MemberMapper.php

```diff
@@ -183,4 +185,44 @@ public function updateMemberNameByUserId(string $userId, string $displayName): v
 			->where($qb->expr()->eq('user_id', $qb->createNamedParameter($userId)));
 		$qb->executeStatement();
 	}
+
+	/**
+	 * Set user_id to null when it points to a user that does not exist
+	 *
+	 * @param string|null $projectId If set, only try to fix members of this project
+	 * @return \Generator
+	 * @throws Exception
+	 */
+	public function cleanupUserMembers(?string $projectId = null): \Generator {
+		$qb = $this->db->getQueryBuilder();
+		$qb->selectDistinct('user_id')
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->isNotNull('user_id')
+			);
+		if ($projectId !== null) {
+			$qb->andWhere(
+				$qb->expr()->eq('project_id', $qb->createNamedParameter($projectId, IQueryBuilder::PARAM_STR))
+			);
+		}
+		$result = $qb->executeQuery();
+
+		$qbUpdate = $this->db->getQueryBuilder();
+		$qbUpdate->update($this->getTableName())
+			->set('user_id', $qbUpdate->createNamedParameter(null))
+			->where(
+				$qbUpdate->expr()->eq('user_id', $qbUpdate->createParameter('nonExistingUserId'))
+			);
+
+		while ($row = $result->fetch()) {
+			$userId = $row['user_id'];
+			if ($this->userManager->get($userId) === null) {
+				$qbUpdate->setParameter('nonExistingUserId', $userId, IQueryBuilder::PARAM_STR);
+				$qbUpdate->executeStatement();
+				yield $userId;
+			}
+		}
+		$result->closeCursor();
```

Comment:
The update part could also be done in one query by collecting all non-existent user IDs first and then updating all members at once since the new value for all is just null.